src/js/letters/actions/letters.js

```diff
@@ -36,29 +36,50 @@ export function getLetterList() {
         data: response,
       }),
       (response) => {
-        const error = response.errors.length > 0 ? response.errors[0] : undefined;
-        if (error) {
-          if (error.status === '503' || error.status === '504') {
+        if (typeof response.errors === 'undefined' || response.errors.length === 0) {
+          return Promise.reject(new Error('vets_letters_error_server_get: undefined error'));
+        }
+        const error = response.errors[0];
+        switch (error.status) {
+          case ('503'): // Handled same as 504
+          case ('504'):
             // Either EVSS or a partner service is down or EVSS times out
             return dispatch({ type: BACKEND_SERVICE_ERROR });
-          }
-          if (error.status === '403') {
+          case ('403'):
             // Backend authentication problem
             return dispatch({ type: BACKEND_AUTHENTICATION_ERROR });
-          }
-          if (error.status === '502') {
-            // Some of the partner services are down, so we cannot verify the eligibility
-            // of some letters
+          case ('502'):
+            // Some of the partner services are down, so we cannot verify the
+            // eligibility of some letters
             return dispatch({ type: LETTER_ELIGIBILITY_ERROR });
-          }
-          return Promise.reject(
-            new Error(`vets_letters_error_server_get: error status ${error.status}`)
-          );
+          default:
+            return Promise.reject(
+              new Error(`vets_letters_error_server_get: ${error.status || 'unknown'}`)
+            );
```

Comment:
Decided to _switch_ this action up a bit 🥁 
Mainly though I noticed when I was making some test cases that there was an unhandled Promise rejection happening when the response had no `errors` property. That's the early return at the top there. The test for this is the 'generic' case below.

Reply:
^ the unhandled promise rejection should be fixed now that `apiRequest()` is being returned from the thunk instead of just called, but I still think it's nicer to break out this error.